### DISCUSSION TITLE
Serverless Functions: a simple example?

### DISCUSSION THREAD

**Discussion by tleylan** [vercel/vercel#5810] [Help] [answered]
I think there must be messages that cover this but I'll ask anyway :-) I'm pretty sure that Vercel can host API services, I believe the recommendation is to use serverless functions is that correct? Could someone recommend a "getting started with example of GET and POST" in Vercel? I have existing code written using KOA? I assume that knowledge would be helpful am I correct that simply moving it into Vercel wouldn't work (or wouldn't work too well)? I'm happy to set up something new I would just like to see it work in an hour or two. Thanks.

**Comment by tleylan** [comment]
I believe I found what I'm looking for. Moving along with this one which seems to illustrate exactly what I needed. https://www.youtube.com/watch?v=BhArBPtW6Ms

**Comment by styfle** [comment, marked as answer]
Hi @tleylan 

The docs have a section called "An Example Node.js Serverless Function" that should get you started.

To deploy a serverless Node.js API, provide a default export function like the following, in a `.js` file within the `/api` directory:

```js
module.exports = (req, res) => {
  res.json({
    body: req.body,
    query: req.query,
    cookies: req.cookies,
  })
}
```

The routing is using the filesystem so if this file is named `/api/index.js` you can invoke it by visiting `/api`.

If its named `/api/foo.js` you can invoke it by visiting `/api/foo`.

https://vercel.com/docs/serverless-functions/introduction

**Reply by styfle** [reply]
Also see this example if you're looking to clone a project: https://github.com/vercel/vercel/tree/master/examples/svelte/api

**Reply by tleylan** [reply]
Excellent thanks. I did get things working but I'm always open to finding more examples. That it is done in Svelte is a plus.
